Tasks/Common/AzureRmDeploy-common/azure-arm-rest/azure-arm-app-service.ts

```diff
@@ -343,6 +372,111 @@ export class AzureAppService {
+        if(force || !this._appServiceConfigurationSettings) {
+            this._appServiceConfigurationSettings = await this._getSlotConfigurationNames();
+        }
+
+        return this._appServiceConfigurationSettings;
+    }
+
+    public async patchConnectionString(addProperties: any, deleteProperties?: any): Promise<any> {
+        var connectionStringSettings = await this.getConnectionStrings(); 
+        var isNewValueUpdated: boolean = false;
+        for(var key in addProperties) {
+            if(JSON.stringify(connectionStringSettings.properties[key]) != JSON.stringify(addProperties[key])) {
+                tl.debug(`Value of ${key} has been changed to ${JSON.stringify(addProperties[key])}`);
+                isNewValueUpdated = true;
+            }
+            connectionStringSettings.properties[key] = addProperties[key];
+        }
+
+        for(var key in deleteProperties) {
+            if(key in connectionStringSettings.properties) {
+                delete connectionStringSettings.properties[key];
+                tl.debug(`Removing app setting : ${key}`);
+                isNewValueUpdated = true;
+            }
+        }
+
+        if(isNewValueUpdated) {
+            await this.updateConnectionStrings(connectionStringSettings);
+        }
+    }
+
+    public async updateConnectionStrings(connectionStringSettings: any): Promise<AzureAppServiceConfigurationDetails> {
+        try {
+            var httpRequest = new webClient.WebRequest();
+            httpRequest.method = 'PUT';
+            httpRequest.body = JSON.stringify(connectionStringSettings);
+            var slotUrl: string = !!this._slot ? `/slots/${this._slot}` : '';
+            httpRequest.uri = this._client.getRequestUri(`//subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/${slotUrl}/config/connectionstrings`,
+            {
+                '{resourceGroupName}': this._resourceGroup,
+                '{name}': this._name,
+            }, null, '2016-08-01');
+            
+            var response = await this._client.beginRequest(httpRequest);
+            if(response.statusCode != 200) {
+                throw ToError(response);
+            }
+
+            return response.body;
+        }
+        catch(error) {
+            throw Error(tl.loc('FailedToUpdateAppServiceConnectionStrings', this._getFormattedName(), this._client.getFormattedError(error)));
+        }
+    }
+
+    public async patchConnectionStringSlot(addProperties: any): Promise<any> {
+        var connectionStringSlotSettings = await this.getSlotConfigurationNames();
+        var isNewValueUpdated: boolean = false;
+        for(var key in addProperties) {
+            if(addProperties[key].slotSetting == true) {
+                if((connectionStringSlotSettings.properties.connectionStringNames.length == 0) || (!connectionStringSlotSettings.properties.connectionStringNames.includes(key))) {
```

Comment:
nit let connectionStringNames = connectionStringSlotSettings.properties.connectionStringNames;
Also is it possible that connectionSTringNames is null or properties is null

Reply:
Already included the check 'connectionStringNames.length == 0'